projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts

```diff
@@ -14,145 +14,184 @@ import {
+  @HostBinding('class.has-components') @Input() hasComponents = false;
 
-  /**
-   * observable with `ContentSlotData` for the current position
-   *
-   * @deprecated we'll stop supporting this property in 2.0 as
-   * it is not used separately.
-   */
-  readonly slot$: Observable<ContentSlotData> = this.position$.pipe(
+  protected position$: BehaviorSubject<string> = new BehaviorSubject(undefined);
+
+  components: ContentSlotComponentData[];
+
+  protected slot$: Observable<ContentSlotData> = this.position$.pipe(
     switchMap((position) => this.cmsService.getContentSlot(position)),
-    tap((slot) => this.addSmartEditSlotClass(slot))
+    distinctUntilChanged(this.isDistinct)
   );
 
-  readonly components$: Observable<
-    ContentSlotComponentData[]
-  > = this.slot$.pipe(
-    map((slot) => (slot && slot.components ? slot.components : [])),
-    distinctUntilChanged(
-      (a, b) =>
-        a.length === b.length && !a.find((el, index) => el.uid !== b[index].uid)
-    )
+  /** Observes the components for the given page slot. */
+  components$: Observable<ContentSlotComponentData[]> = this.slot$.pipe(
+    map((slot) => slot?.components ?? [])
   );
 
-  private subscription = new Subscription();
+  protected subscription: Subscription = new Subscription();
 
-  constructor(
-    cmsService: CmsService,
-    dynamicAttributeService: DynamicAttributeService,
-    renderer: Renderer2,
-    hostElement: ElementRef,
-    // tslint:disable-next-line:unified-signatures
-    config: CmsConfig
-  );
-  /**
-   * @deprecated since version 1.4
-   * Use constructor(cmsService: CmsService, dynamicAttributeService: DynamicAttributeService, renderer: Renderer2, hostElement: ElementRef, config?: CmsConfig) instead
-   */
-  constructor(
-    cmsService: CmsService,
-    dynamicAttributeService: DynamicAttributeService,
-    renderer: Renderer2,
-    hostElement: ElementRef
-  );
+  /** Keeps track of the pending components that must be loaded for the page slot */
+  private pendingComponentCount = 0;
+
+  /** Tracks the last used position, in case the page slot is used dynamically */
+  private lastPosition: string;
   constructor(
     protected cmsService: CmsService,
     protected dynamicAttributeService: DynamicAttributeService,
     protected renderer: Renderer2,
-    protected hostElement: ElementRef,
-    protected config?: CmsConfig,
-    protected cdRef?: ChangeDetectorRef
+    protected elementRef: ElementRef,
+    protected config: CmsConfig,
+    protected cd: ChangeDetectorRef
   ) {}
 
   ngOnInit() {
     this.subscription.add(
-      this.components$.subscribe((components) => {
-        this.hasComponents = components && components.length > 0;
-        this.pendingComponentCount = components ? components.length : 0;
-        this.isPending = this.pendingComponentCount > 0;
-      })
+      this.slot$
+        .pipe(tap((slot) => this.decorate(slot)))
+        .subscribe((value) => (this.components = value?.components || []))
```

Comment:
As we don't have async pipe anywhere inside, I'm wondering if addin cd.markForCheck() here won't be needed in some cases?

Reply:
Good point. Not happing in our code base, but could happen if the page slot component is used in a situation where the slot data has not been loaded upfront.